src/tests/gc-e2e/GuestConfiguration.Tests.ps1

```diff
@@ -0,0 +1,42 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# GuestConfiguration.Tests.ps1
+param (
+    [Parameter(Mandatory)]
+    [string] $PolicyPackage
+)
+
+Describe 'Validate GuestConfiguration NRP' {
+    # Perform monitoring - Get
+    Context "Get" {
+        BeforeAll {
+            $result = Get-GuestConfigurationPackageComplianceStatus -Path $PolicyPackage -Verbose
+        }
+
+        It 'Correct resource count - 20' {    
+            $result | Should -Not -BeNullOrEmpty
+            $result.resources | Should -HaveCount 20
+        }
```

Comment:
We need to validate more from the returned results. More than just number of instances. Can we get back the pass results and the reasons and validate  pass and non empty reasons strings for each instance?